drivers/input/input_cf1133.c

```diff
@@ -0,0 +1,367 @@
+	/** Device pointer. */
+	const struct device *dev;
+	/** Work queue (for deferred read). */
+	struct k_work work;
+#ifdef CONFIG_INPUT_CF1133_INTERRUPT
+	/** Interrupt GPIO callback. */
+	struct gpio_callback int_gpio_cb;
+#else
+	/** Timer (polling mode). */
+	struct k_timer timer;
+#endif
+	/** Last pressed state. */
+	bool pressed_old;
+	bool pressed;
+
+	int resolution_x;
+	int resolution_y;
+	uint8_t max_touches;
+	uint8_t touch_protocol_type;
+	uint8_t pixel_length;
+	uint8_t chip_id;
+
+	uint8_t Num_X;
+	uint8_t Num_Y;
+	uint8_t sensing_mode;
+};
+
+static int cf1133_get_chip_id(struct cf1133_data *data, const struct cf1133_config *config)
+{
+	int ret = 0;
+	uint8_t buffer[3];
+
+	ret = i2c_burst_read_dt(&config->bus, CHIP_ID, buffer, sizeof(buffer));
+	if (ret < 0) {
+		LOG_ERR("Read burst failed");
+		return ret;
+	}
+	if (buffer[0] == 0) {
+		if (buffer[1] + buffer[2] > 32) {
+			data->chip_id = 2;
+		} else {
+			data->chip_id = 0;
+		}
+	} else {
+		data->chip_id = buffer[0];
+	}
+	data->Num_X = buffer[1];
+	data->Num_Y = buffer[2];
+	LOG_DBG("Chip ID = %d", data->chip_id);
+	LOG_DBG("Num_X = %d", data->Num_X);
+	LOG_DBG("Num_Y = %d", data->Num_Y);
+
+	return 0;
+}
+
+static int cf1133_get_protocol_type(struct cf1133_data *data, const struct cf1133_config *config)
+{
+	int ret = 0;
+	uint8_t buffer;
+
+	if (data->chip_id <= 3) {
+		ret = i2c_reg_read_byte_dt(&config->bus, I2C_PROTOCOL, &buffer);
+		if (ret < 0) {
+			LOG_ERR("read i2c protocol error (%d)", ret);
+			return ret;
+		}
+		data->touch_protocol_type = buffer & I2C_PROTOCOL_BMSK;
+		LOG_DBG("i2c protocol = %d ", data->touch_protocol_type);
+		data->sensing_mode =
+			(buffer & (ONE_D_SENSING_CONTROL_BMSK << ONE_D_SENSING_CONTROL_SHFT)) >>
+			ONE_D_SENSING_CONTROL_SHFT;
+		LOG_DBG("sensing mode = %d ", data->sensing_mode);
+	} else {
+		data->touch_protocol_type = SITRONIX_A_TYPE;
+		LOG_DBG("i2c protocol = %d ", data->touch_protocol_type);
+
+		ret = i2c_reg_read_byte_dt(&config->bus, 0xf0, &buffer);
+		if (ret < 0) {
+			LOG_ERR("read sensing mode error (%d)", ret);
+			return ret;
+		}
+		data->sensing_mode = (buffer & ONE_D_SENSING_CONTROL_BMSK);
+		LOG_DBG("sensing mode = %d", data->sensing_mode);
+	}
+	return 0;
+}
+
+static int cf1133_get_max_touches(struct cf1133_data *data, const struct cf1133_config *config)
+{
+	int ret = 0;
+	uint8_t buffer;
+
+	ret = i2c_reg_read_byte_dt(&config->bus, MAX_NUM_TOUCHES, &buffer);
+	if (ret < 0) {
+		LOG_ERR("read sensing mode error (%d)", ret);
+		return ret;
+	}
+	data->max_touches = buffer;
+	if (data->max_touches > SITRONIX_MAX_SUPPORTED_POINT) {
+		data->max_touches = SITRONIX_MAX_SUPPORTED_POINT;
+	}
+	LOG_DBG("max touches = %d ", data->max_touches);
+
+	return 0;
+}
+
+static int cf1133_ts_init(struct cf1133_data *data, const struct cf1133_config *config)
+{
+	int r;
+	/* Get device status before use to do once */
+	r = cf1133_get_chip_id(data, config);
+	if (r < 0) {
+		LOG_ERR("Read chip id failed error %d :", r);
```

Comment:
```suggestion
		LOG_ERR("Read chip id failed: %d", r);
```

Same on the other ones. You could also use a generic message and keep it the same on the others, save a bit of flash, up to you.